Source/NSCell.m

```diff
@@ -2966,31 +2979,70 @@ - (NSSize) _sizeText: (NSString*)title
 - (NSAttributedString*) _drawAttributedString
 {
   if (!_cell.is_disabled)
-    {  
+    {
       return [self attributedStringValue];
     }
   else
     {
       NSAttributedString *attrStr = [self attributedStringValue];
       NSDictionary *attribs;
       NSMutableDictionary *newAttribs;
-              
-      attribs = [attrStr attributesAtIndex: 0 
+
+      attribs = [attrStr attributesAtIndex: 0
                          effectiveRange: NULL];
       newAttribs = [NSMutableDictionary dictionaryWithDictionary: attribs];
       [newAttribs setObject: [NSColor disabledControlTextColor]
                   forKey: NSForegroundColorAttributeName];
-      
+
       return AUTORELEASE([[NSAttributedString alloc]
                              initWithString: [attrStr string]
                              attributes: newAttribs]);
     }
 }
 
+
+- (BOOL) _shouldShortenStringForRect: (NSRect)titleRect size: (NSSize)titleSize length: (NSUInteger)length
+{
+  NSLineBreakMode mode = [self lineBreakMode];
+
+  return ((titleSize.width > titleRect.size.width) && (length > 4) &&
+          (mode == NSLineBreakByTruncatingHead ||
+           mode == NSLineBreakByTruncatingTail ||
+           mode == NSLineBreakByTruncatingMiddle));
+}
+
+- (NSAttributedString*) _resizeAttributedString: (NSAttributedString*)attrstring forRect: (NSRect)titleRect
+{
+  // Redo string based on selected truncation mask...
+  NSMutableAttributedString *mutableString = AUTORELEASE([attrstring mutableCopy]);
+  NSString *ellipsis = @"...";
+  NSLineBreakMode mode = [self lineBreakMode];
+  // This code shortens the string one character at a time.
+  // To speed it up we start off propertional:
```

Comment:
What's `propertional`? If it means `proportional`, I don't think this code uses proportional size, it seems to start with fixed character sizes.

Reply:
Yes, this should be "proportional". The idea is to cut off as many characters as the width of the string exceeds the width of the rectangle. In German this is called Dreisatz, which translates into "rule of proportion".

Reply:
Yes, my mistake was not seeing that `[mutableString size].width` will return proportional size. Oops!